packages/expo-cli/src/commands/upgrade.ts

```diff
@@ -0,0 +1,217 @@
+import { Command } from 'commander';
+import { Versions } from '@expo/xdl';
+import JsonFile from '@expo/json-file';
+import * as ConfigUtils from '@expo/config';
+import chalk from 'chalk';
+import * as PackageManager from '../PackageManager';
+import CommandError from '../CommandError';
+import prompt from '../prompt';
+import log from '../log';
+import path from 'path';
+import fs from 'fs';
+
+type Options = {
+  npm?: boolean;
+  yarn?: boolean;
+};
+
+function maybeFormatSdkVersion(sdkVersionString: string | null) {
+  if (!sdkVersionString) {
+    return sdkVersionString;
+  } else if (sdkVersionString.includes('.')) {
+    return sdkVersionString;
+  }
+
+  return `${sdkVersionString}.0.0`;
+}
+
+type DependencyList = { [key: string]: string };
+
+/**
+ * Produce a list of dependencies used by the project that need to be updated
+ */
+async function getUpdatedDependenciesAsync(
+  projectRoot: string,
+  targetSdkVersion: { expoReactNativeTag: string; facebookReactVersion?: string } | null
+): Promise<DependencyList> {
+  let result: DependencyList = {};
+
+  // Get the supported react-native version.
+  // * We can't update the react-native version if the version data isn't published
+  if (targetSdkVersion) {
+    result['react-native'] = `https://github.com/expo/react-native/archive/${
+      targetSdkVersion.expoReactNativeTag
+    }.tar.gz`;
+
+    // React version apparently is optional in SDK version data
+    if (targetSdkVersion.facebookReactVersion) {
+      result['react'] = targetSdkVersion.facebookReactVersion;
+    }
+
+    // - TODO: how do we know what version of react-native-web, react-dom to use?
+    // - TODO: we need to track supported devDependencies versions somewhere programmatic too
+  } else {
+    console.log(`Supported React Native version unknown, please update it manually.`);
+  }
+
+  // Get the updated version for any bundled modules
+  let { exp, pkg } = await ConfigUtils.readConfigJsonAsync(projectRoot);
+  let bundledNativeModules = (await JsonFile.readAsync(
+    ConfigUtils.resolveModule('expo/bundledNativeModules.json', projectRoot, exp)
+  )) as DependencyList;
+
+  let { dependencies } = pkg;
+  Object.keys(bundledNativeModules).forEach(name => {
+    if (dependencies[name]) {
+      result[name] = bundledNativeModules[name];
+    }
+  });
+
+  return result;
+}
+
+async function upgradeAsync(requestedSdkVersion: string | null, options: Options) {
+  let { projectRoot, workflow } = await findProjectRootAsync(process.cwd());
+  let { exp } = await ConfigUtils.readConfigJsonAsync(projectRoot);
+
+  if (Versions.lteSdkVersion(exp, '32.0.0')) {
```

Comment:
```suggestion
  if (!Versions.gteSdkVersion(exp, '33.0.0')) {
```
(I know we currently don't have versions like `33.0.1`, but we should probably still treat the version number like a semver.)